forem/forem
# Round Robin notifications design refresh

## What type of PR is this? (check all applicable)


## Description

The notifications that are sent to trusted users/moderators when a new user comments needed a clearer CTA and an option to opt out in the notifications page.

## Related Tickets & Documents



- Related Issue #19296 
- Closes #19296 

## QA Instructions, Screenshots, Recordings

<details>
<summary>QA Procedure (click to expand)</summary>

- **Important!** Before pulling this branch, create an "old style" notification
  - Create a new user account
  - Comment on an article with it
  - This will send a mod notification to one of your trusted users; note which one (likely it's just the admin)
  - Visit their notifications page to view it:
  -  <img width="751" alt="Old notification with old style" src="https://user-images.githubusercontent.com/29291843/235905684-9e3b8e74-8b89-4a1b-b9b3-e832f1d44b89.png">

- After pulling the branch
  - Reload the trusted user's notification page; we want to be sure that old/existing notifications don't break
  - The notification should render in the new style, but still with the commenter's username rather than their name
  - <img width="745" alt="Old notification with new style" src="https://user-images.githubusercontent.com/29291843/235905854-08031bee-75c5-4874-a22a-af879b6bdef9.png">
  - Confirm that you can interact with the notification as normal (react, reply, etc)
  - Confirm that the "Change settings" link goes to the notifications settings page

- Test out new round robin notifications:
  - Create a new trusted user account (to ensure you have at least one user that can receive round robin notifications)
    - Create a new user account
    - React to an article with it
    - Using your admin account, mark the new user as trusted via the Member Manager in the admin section
  - Create another new user account and comment on an article with it
  - This will send a mod notification to one of your trusted users; visit their notifications page to view it:
  - <img width="747" alt="New notification with new style" src="https://user-images.githubusercontent.com/29291843/235905871-aec6412f-8494-4c7a-93dc-2c4cda4380da.png">
  - Same checks for notification interactivity as before

🎉 

</details>

### UI accessibility concerns?

N/A (Largely just changing copy - I did add a link but it is a proper link)

## Added/updated tests?

- [x] Yes, some unit tests and updated a request spec
      have not been included_

## [optional] Are there any post deployment tasks we need to perform?

Not as far as I know - this implementation should handle both existing and newly created notifications
